source/common/config/decoded_resource_impl.h

```diff
@@ -77,15 +77,15 @@ class DecodedResourceImpl : public DecodedResource {
   bool hasResource() const override { return has_resource_; }
   absl::optional<std::chrono::milliseconds> ttl() const override { return ttl_; }
   const OptRef<const envoy::config::core::v3::Metadata> metadata() const override {
-    return metadata_;
+    return metadata_.has_value() ? makeOptRef(metadata_.value()) : absl::nullopt;
   }
 
 private:
   DecodedResourceImpl(OpaqueResourceDecoder& resource_decoder, absl::optional<std::string> name,
                       const Protobuf::RepeatedPtrField<std::string>& aliases,
                       const ProtobufWkt::Any& resource, bool has_resource,
                       const std::string& version, absl::optional<std::chrono::milliseconds> ttl,
-                      const OptRef<const envoy::config::core::v3::Metadata> metadata)
+                      const absl::optional<envoy::config::core::v3::Metadata> metadata)
```

Comment:
nit: I'm never sure if this style of passing a non-pod by const value will result in an extra copy.

Consider passing by rvalue-ref or const ref.